### ISSUE TITLE
Custom key format?

### ISSUE THREAD

**Issue by Flithor** [dotnet/ResXResourceManager#398]
I needs to move text to resource with a special format key( Namespace.TypeName.Method.Kind.Key)
Is there any config that allows me to do this?

**Comment by tom-englert** [comment]
Yes, there are up to three variants available, just select the desired format:

![image](https://user-images.githubusercontent.com/10229954/112821489-0ee7f780-9087-11eb-82c5-bd575219a8c7.png)

**Comment by Flithor** [comment]
If I want to use a custom key format like `Namespace.Type.Method.Kind.Key`?

**Comment by tom-englert** [comment]
How would you use this in code? Values with dots are not legal identifiers. 
You could not write
```c#
var text = Resources.Namespace.Type.Method.Key;
```
but
```c#
var text = Resources.Namespace_Type_Method_Key;
```

**Comment by Flithor** [comment]
Same as other issue: indexer

**Comment by send0xx** [comment]
Is there any way to bypass/improve this validation?
`Namespace.Type` convention is used and project has no code generation, 
Resources can be get by string identifier via `ResourceManager` (even with dot notation)

Also VS2022 designer converts `.` to `_`
![image](https://github.com/dotnet/ResXResourceManager/assets/31574812/c783dc74-2036-4960-a684-fa4b031affec)

**Comment by tom-englert** [comment]
Using dots in keys generates conflicts in so many places, it's better to go without them.
Anyhow I don't understand why everybody loves dots and hates underscores.
